doc/source/cluster/kubernetes/examples/rayserve-llm-example.md

```diff
@@ -0,0 +1,153 @@
+(kuberay-rayservice-llm-example)=
+
+# Serve a Large Language Model using Ray Serve LLM on Kubernetes
+
+This guide provides a step-by-step walkthrough for deploying a Large Language Model (LLM) using Ray Serve LLM on Kubernetes. Leveraging KubeRay, Ray Serve, and vLLM, this guide deploys the  `Qwen/Qwen2.5-7B-Instruct` model from HuggingFace, enabling scalable, efficient, and OpenAI-compatible LLM serving within a Kubernetes environment.
+
+## Prerequisites
+
+In this example, model weights are downloaded from Hugging Face. In order to properly finish this guide, you must fulfill the following requirements:
+* A [Hugging Face Account](https://huggingface.co/) and a Hugging Face [Access Token](https://huggingface.co/settings/tokens) with read access to gated repos.
+* In your Ray Serve configuration, set the `HUGGING_FACE_HUB_TOKEN` environment variable to this token to enable model downloads.
+
+## Install the KubeRay Operator
+
+Install the most recent stable KubeRay operator from the Helm repository by following [Deploy a KubeRay operator](kuberay-operator-deploy). The KubeRay operator Pod must be on the CPU node if you set up the taint for the GPU node pool correctly.
+
+
+## Creating a Namespace for Ray Clusters
+
+Create a Kubernetes Namespace for serving Ray Serve Applications:
+
+```sh
+kubectl create namespace ray-serve
+```
+
+## Create a Kubernetes Secret containing your Hugging Face access token
+
+Additionally, instead of passing the HF Access Token directly as an environment variable, you can also create a Kubernetes Secret containing your Hugging Face access token:
+
+```yaml
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    name: hf-token
+  type: Opaque
+  stringData:
+    hf_token: <your-hf-access-token-value>
+```
+
+```sh
+kubectl apply -f hf-token-secret.yaml -n ray-serve
```

Comment:
No need to have another YAML. We can put everything (e.g. Secret, RayService) in a single YAML.